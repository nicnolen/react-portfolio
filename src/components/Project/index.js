import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';

function Project({ title, description, image, github, deployed, tools }) {
  return (
    <div className='card'>
      <div className='img-container'>
        <a href={deployed}>
          <img src={image} alt={description} width='150' height='70' />
        </a>
      </div>
      <div className='content'>
        <a href={github} title={description}>
          <img
            src='../../assets/images/projectImages/github.png'
            alt='GitHub Repo'
          />
          <h3 className='card-title'>{title}</h3>
        </a>
        <h4>{tools}</h4>
      </div>
    </div>
  );
}

export default Project;
