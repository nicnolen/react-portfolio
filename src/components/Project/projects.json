[
  {
    "id": "1",
    "title": "Travel Lovers",
    "description": "Professional portfolio to showcase skills to employers",
    "image": "../../assets/images/projectImages/travel-lovers.png",
    "github": "https://github.com/nicnolen/Travel-Lovers",
    "deploy": "https://travel--lovers.herokuapp.com/",
    "tools": "Handlebars.js, Sequelize, SASS, bcrypt"
  },
  {
    "id": "2",
    "title": "Travel Lovers",
    "description": "Professional portfolio to showcase skills to employers",
    "image": "../../assets/images/projectImages/travel-lovers.png",
    "github": "https://github.com/nicnolen/Travel-Lovers",
    "deploy": "https://travel--lovers.herokuapp.com/",
    "tools": "Handlebars.js, Sequelize, SASS, bcrypt"
  }
]
