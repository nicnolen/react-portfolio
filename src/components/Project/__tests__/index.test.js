// Import dependencies
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Project from '..';

// After each test, make sure there isnt any leftover memory that could give you false results
afterEach(cleanup);

// Declare the components you are testing
describe('Portfolio component', () => {
  // verify the component is rendering properly
  it('Portfolio component renders', () => {
    render(<Project />);
  });

  // compare snapshot versions of the DOM node structure
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Project />);
    expect(asFragment()).toMatchSnapshot();
  });
});