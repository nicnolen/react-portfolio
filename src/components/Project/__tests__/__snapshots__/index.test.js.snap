// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="card mb-5"
  >
    <div
      class="description"
    />
    <figure
      class="img-container mb-0"
    >
      <a>
        <img
          class="project"
        />
      </a>
    </figure>
    <div
      class="content d-flex justify-content-between align-items-center"
    >
      <span
        class="github-link"
      >
        <a>
          <img
            alt="GitHub Repo"
            src="http://nicnolen.github.io/react-portfolio/assets/images/icons/github.svg"
          />
          <h3
            class="card-title"
          />
        </a>
      </span>
      <h4
        class="project-skills fs-4 pb-1 px-1"
      />
    </div>
  </div>
</DocumentFragment>
`;
